Plugin~/WebRTCPlugin/UnityRenderEvent.cpp

```diff
@@ -142,3 +167,58 @@ extern "C" UnityRenderingEventAndData UNITY_INTERFACE_EXPORT UNITY_INTERFACE_API
     s_context = context;
     return OnRenderEvent;
 }
+
+static void UNITY_INTERFACE_API TextureUpdateCallback(int eventID, void* data)
+{
+    auto event = static_cast<UnityRenderingExtEventType>(eventID);
+
+    if (event == kUnityRenderingExtEventUpdateTextureBeginV2)
+    {
+        auto params = reinterpret_cast<UnityRenderingExtTextureUpdateParamsV2 *>(data);
+        
+        auto renderer = s_context->GetVideoRenderer(params->userData);
+        if (renderer == nullptr)
+        {
+            DebugLog("VideoRenderer not found, rendererId:%d", params->userData);
+            return;
+        }
+
+        auto frame = renderer->GetFrameBuffer();
+        if(frame == nullptr)
+        {
+            DebugLog("VideoFrame is not received yet, rendererId:%d", params->userData);
+            return;
+        }
+
+        rtc::scoped_refptr<webrtc::I420Buffer> i420_buffer = webrtc::I420Buffer::Create(params->width, params->height);
+        i420_buffer->ScaleFrom(*frame->ToI420());
+        delete[] renderer->tempBuffer;
+        renderer->tempBuffer = new uint8_t[params->width * params->height * 4];
```

Comment:
Shouldn't you be going to use smart pointer?

Reply:
I tried smartpointer(unique_ptr), but it didn't work.
Perhaps, If it is a smart pointer, it seems to be released once the Rendering thread is out of processing.

Reply:
I use `vector<uint8_t>`, thanks!